internal/controller/policyserver_controller_test.go

```diff
@@ -247,32 +247,195 @@ var _ = Describe("PolicyServer controller", func() {
+				ContextAwareResources: admissionPolicy.GetContextAwareResources(),
+				Policies:              buildPoliciesMembers(admissionPolicy.GetPolicyMembers()),
+				Expression:            admissionPolicy.GetExpression(),
+				Message:               admissionPolicy.GetMessage(),
 			}
+			policiesMap[clusterAdmissionPolicy.GetUniqueName()] = policyServerConfigEntry{
+				NamespacedName: types.NamespacedName{
+					Namespace: clusterAdmissionPolicy.GetNamespace(),
+					Name:      clusterAdmissionPolicy.GetName(),
+				},
+				URL:                   clusterAdmissionPolicy.GetModule(),
+				PolicyMode:            string(clusterAdmissionPolicy.GetPolicyMode()),
+				AllowedToMutate:       clusterAdmissionPolicy.IsMutating(),
+				Settings:              clusterAdmissionPolicy.GetSettings(),
+				ContextAwareResources: clusterAdmissionPolicy.GetContextAwareResources(),
+				Policies:              buildPoliciesMembers(clusterAdmissionPolicy.GetPolicyMembers()),
+				Expression:            clusterAdmissionPolicy.GetExpression(),
+				Message:               clusterAdmissionPolicy.GetMessage(),
+			}
+			policiesMap[admissionPolicyGroup.GetUniqueName()] = policyServerConfigEntry{
+				NamespacedName: types.NamespacedName{
+					Namespace: admissionPolicyGroup.GetNamespace(),
+					Name:      admissionPolicyGroup.GetName(),
+				},
+				URL:                   admissionPolicyGroup.GetModule(),
+				PolicyMode:            string(admissionPolicyGroup.GetPolicyMode()),
+				AllowedToMutate:       admissionPolicyGroup.IsMutating(),
+				Settings:              admissionPolicyGroup.GetSettings(),
+				ContextAwareResources: admissionPolicyGroup.GetContextAwareResources(),
+				Policies:              buildPoliciesMembers(admissionPolicyGroup.GetPolicyMembers()),
+				Expression:            admissionPolicyGroup.GetExpression(),
+				Message:               admissionPolicyGroup.GetMessage(),
+			}
+			policiesMap[clusterPolicyGroup.GetUniqueName()] = policyServerConfigEntry{
+				NamespacedName: types.NamespacedName{
+					Namespace: clusterPolicyGroup.GetNamespace(),
+					Name:      clusterPolicyGroup.GetName(),
+				},
+				URL:                   clusterPolicyGroup.GetModule(),
+				AllowedToMutate:       clusterPolicyGroup.IsMutating(),
+				Settings:              clusterPolicyGroup.GetSettings(),
+				ContextAwareResources: clusterPolicyGroup.GetContextAwareResources(),
+				PolicyMode:            string(clusterPolicyGroup.GetPolicyMode()),
+				Policies:              buildPoliciesMembers(clusterPolicyGroup.GetPolicyMembers()),
+				Expression:            clusterPolicyGroup.GetExpression(),
+				Message:               clusterPolicyGroup.GetMessage(),
+			}
+
 			policies, err := json.Marshal(policiesMap)
 			Expect(err).ToNot(HaveOccurred())
 
+			// As we have a customization in how we serialize the policies,
+			// let's check if the result json is in the expected format.
+			// Otherwise, the policy server will not start.
```

Comment:
Would it be more readable to have the custom marshaller/unmarshaller tested inside of a traditional (as in, no ginko) test? Maybe introduce a `policyserver_controller_configmap_test.go`

Reply:
I'm ok with it being a Ginkgo test.